planner/cascades/testdata/transformation_rules_suite_out.json

```diff
@@ -234,6 +234,29 @@
           "Group#10 Schema:[test.t.c,test.t.d]",
           "    IndexScan_24 table:t, index:c, d, e, cond:[gt(test.t.c, 1)]"
         ]
+      },
+      {
+        "SQL": "select a, b from (select a, b, min(a) over(partition by b) as min_a from t)as tt where a < 10 and b > 10 and b = min_a",
+        "Result": [
+          "Group#0 Schema:[test.t.a,test.t.b]",
+          "    Projection_7 input:[Group#1], test.t.a, test.t.b",
+          "Group#1 Schema:[test.t.a,test.t.b,Column#14]",
+          "    Projection_5 input:[Group#2], test.t.a, test.t.b, Column#14",
+          "Group#2 Schema:[test.t.a,test.t.b,Column#14]",
+          "    Selection_12 input:[Group#3], eq(test.t.b, Column#14), lt(test.t.a, 10)",
+          "Group#3 Schema:[test.t.a,test.t.b,Column#14]",
+          "    Window_4 input:[Group#4]",
+          "Group#4 Schema:[test.t.a,test.t.b]",
+          "    Projection_3 input:[Group#5], test.t.a, test.t.b",
```

Comment:
I'm not sure whether it's right. For the `Projection_3` may be unexpected.
And I wonder why we can only push b > 10 down to window, that is why we can only push col in partition by down to window?

Reply:
Let me explain the second one first.
Push the filters which contain the column not in partition by clause would cause wrong result.
Take the given case as example. There're two rows `a=-1 and b = 11` and `a = 10 and b = 11`. If we push the filter `a < 10` down to the window. `a=-1 and b=11` will be filtered. Then the partition `b=11`'s `min(a)` would become `10` instead of `-1`.

Reply:
And for the first one.
Though windown is in the select field but it's calculated at last.
For a SQL pattern `select normal_field, window_field from tables where filters`:
The `tables` in FROM CLAUSE is calcualted first, then is the normal_field, then filters, then window_field. So some redundant project may be added when build plans. We'll use the EliminateProjection rule to eliminate the unnecessary ones.

Reply:
OK, Thank you for the awesome explanation. But my github name is @jiangyuzhao , hhh.